### PR TITLE
feat: add support for PHP 8.3 typed class constants

### LINKED ISSUE #1133: Fails to parse classes with typed constants
It looks like files that have typed constants are not parseable by the latest version (3.1.5).

Input file:
```
<?php

class Test {
   private const int MY_CONST = 3;
}
```
Error:
```
SyntaxError: Parse Error : syntax error, unexpected 'MY_CONST' (T_STRING), expecting '=' on line 4
    at Parser.raiseError (/home/user/Code/node_modules/.pnpm/php-parser@3.1.5/node_modules/php-parser/src/parser.js:349:17)
    at Parser.error (/home/user/Code/node_modules/.pnpm/php-parser@3.1.5/node_modules/php-parser/src/parser.js:397:15)
    at Parser.expect (/home/user/Code/node_modules/.pnpm/php-parser@3.1.5/node_modules/php-parser/src/parser.js:577:10)
    at Parser.read_constant_declaration (/home/user/Code/node_modules/.pnpm/php-parser@3.1.5/node_modules/php-parser/src/parser/class.js:261:18)
    at Parser.read_list (/home/user/Code/node_modules/.pnpm/php-parser@3.1.5/node_modules/php-parser/src/parser/utils.js:76:33)
    at Parser.read_constant_list (/home/user/Code/node_modules/.pnpm/php-parser@3.1.5/node_modules/php-parser/src/parser/class.js:237:24)
    at Parser.read_class_body (/home/user/Code/node_modules/.pnpm/php-parser@3.1.5/node_modules/php-parser/src/parser/class.js:117:32)
    at Parser.read_class_declaration_statement (/home/user/Code/node_modules/.pnpm/php-parser@3.1.5/node_modules/php-parser/src/parser/class.js:32:30)
    at Parser.read_top_statement (/home/user/Code/node_modules/.pnpm/php-parser@3.1.5/node_modules/php-parser/src/parser/statement.js:52:21)
    at Parser.read_start (/home/user/Code/node_modules/.pnpm/php-parser@3.1.5/node_modules/php-parser/src/parser/main.js:18:19) {
  lineNumber: 4,
  fileName: 'eval',
  columnNumber: 21
}
```

### PR DESCRIPTION
**Pull request by Kenneth-Sills**
[Related RFC](https://wiki.php.net/rfc/typed_class_constants).

Closes #1133

### CODE REVIEW

**Comment by Kenneth-Sills** [comment]
@mreiden I noticed you forked from my branch here. If you noticed a bug or have a better approach, would you mind commenting here so the maintainers can be aware if they pick up the review? Thank you! :smile:

**Review by ichiriac — APPROVED** [review]
👍

**Review by czosel — APPROVED** [review]
LGTM as well, sorry about the massive delay and thanks for contributing @Kenneth-Sills!

File: src/parser/class.js [hunk 1]
```diff
@@ -226,13 +226,17 @@ module.exports = {
...
+   *  constant_list ::= T_CONST [type] (constant_declaration ',')* constant_declaration
    * ```
    */
   read_constant_list: function (flags, attrs) {
     if (this.expect(this.tok.T_CONST)) {
       this.next();
     }
+
+    const [nullable, type] =
+      this.version >= 830 ? this.read_optional_type() : [false, null];
```
Comment (genintho) [inline src/parser/class.js:238]: I believe PHP 8.3 translate to version "803". See [this comment ](https://github.com/glayzzle/php-parser/blob/main/src/index.js#L49) and the [implementation of version parsing](https://github.com/glayzzle/php-parser/blob/main/src/index.js#L87).

Comment (czosel) [reply src/parser/class.js:238]: Indeed, thanks for pointing that out! Fix: https://github.com/glayzzle/php-parser/pull/1149

**Pull request merged**